### PR TITLE
Faster Element Symbol Handling

### PR DESCRIPTION
**Pull request by johnmay**
- Replaces lookup of const atomic symbols with the existing constant definitions.
- Add some extra EL_NUMBER_ constants
- Optimise the lookup for common cases
- Replace lazy initialisation of some element arrays with a much simpler and less error prone declaration.

Testing on ChEMBL I see 2.4 million identical InChI's and runs ~9 seconds faster (M1 Pro).
Overall I mainly was wincing and some of the lookups inside nested ternaries (e.g. ichitaut.c)

```
main:
INCHI_EXE/bin/Linux/inchi-1 /data/chembl_34.sdf -AuxNone -NoWarnings -NoLabel
264.73s user 24.96s system 99% cpu 4:51.59 total
PR:
INCHI_EXE/bin/Linux/inchi-1 /data/chembl_34.sdf -AuxNone -NoWarnings -NoLabel
257.91s user 23.88s system 99% cpu 4:42.53 total
```

I may have left some unused variables when I removed the lazy ``if (!en){ }`` initialisations but hopefully got them all.

### CODE REVIEW

**Review by JanCBrammer — COMMENTED** [review]
Trading interoperability with extraterrestrial periodic tables for simplicity sounds good to me :alien: 

* Ran CI locally on `elname-optimisations`, all tests pass (CI has problems on GitHub runners due to access-rights from forks: https://github.com/IUPAC-InChI/InChI/actions/runs/10079959826).

* Checked for left-over uses of `get_periodic_table_number()` that could be replaced with a static declaration. As far as I can tell, there's one more opportunity for static declaration?
https://github.com/IUPAC-InChI/InChI/blob/c94cf244c926f89408289922656ff34614480770/INCHI-1-SRC/INCHI_BASE/src/ichirvr1.c#L749

**Comment by johnmay** [comment]
I've made some more changes to where the element arrays get setup in various places (mainly strutil.c ion_pair handling). I also found some bugs from when the thread safety was added, nothing major but some had an out of bounds checks which wouldn't have been obvious to the static analysis.

**Comment by djb-rwth** [comment]
Hi @johnmay,
Thank you for these useful suggestions.
These changes make sense as using constants instead of calling a function can indeed make a notable increase in performance, especially whilst processing large datasets.

The only suggestion from my side is to add these changes to the `rwth` branch instead of `main` directly as I need to add a file which should fix a rather serious bug; afterwards everything shall be merged into the `main` branch.

File: INCHI-1-SRC/INCHI_BASE/src/ichi_bns.c [hunk 1]
```diff
@@ -2185,15 +2185,21 @@ int fix_special_bonds( BN_STRUCT *pBNS,
...
+        EL_NUMBER_O,
+        EL_NUMBER_S,
+        EL_NUMBER_SE,
+        EL_NUMBER_TE,
+        EL_NUMBER_C
+#if ( FIX_REM_ION_PAIRS_Si_BUG == 1 )        
+        ,EL_NUMBER_SI
+#endif        
+    };
+    static int ne = sizeof(en)/sizeof(en[0]);
```
[the remark this answers is not in the document]
Comment (djb-rwth) [reply INCHI-1-SRC/INCHI_BASE/src/ichi_bns.c:2202]: Having this as another patch would be better and easier to review.

**Review by djb-rwth — COMMENTED** [review]
All good.

**Pull request merged**